src/osal/files_win32.c

```diff
@@ -91,8 +93,10 @@ int osal_mkdirp(const char *dirpath, int mode)
     if (strchr(OSAL_DIR_SEPARATORS, *lastchar) != NULL)
         *lastchar = 0;
 
+    wchar_t w_mypath[PATH_MAX];
+    MultiByteToWideChar(CP_UTF8, 0, mypath, -1, w_mypath, PATH_MAX);
     // Terminate quickly if the path already exists
-    if (_stat(mypath, &fileinfo) == 0 && (fileinfo.st_mode & _S_IFDIR))
+    if (_wstat(w_mypath, &fileinfo) == 0 && (fileinfo.st_mode & _S_IFDIR))
         goto goodexit;
 
     while ((currpath = strpbrk(currpath + 1, OSAL_DIR_SEPARATORS)) != NULL)
```

Comment:
Looking over this function in particular, I think this loop will not work with your wide-char modifications.  It is going through the path components and modifying the original (utf8) string, but you are calling _wstat and _wmkdir on the wide-char converted string.  For this to work properly, the currpath and mypath variables should be operating on the wide-char string, not the original UTF8 string.

Reply:
Thanks, yeah when I tested this I only did it with an existing directory, I didn't look at this close enough.

I fixed up osal_mkdirp to use wide chars from start to finish. I tested on my PC by settings my config directory to "existing_dir/one/two" and it did create directories "one" and "two" and put the config inside there.